### ISSUE TITLE
IntegrityCheckFailed when running Bun in Docker

### ISSUE THREAD

**Issue by ilrein** [oven-sh/bun#5581]
### What version of Bun is running?

1.0.2+37edd5a6e389265738e89265bcbdf2999cb81a49

### What platform is your computer?

Darwin 22.6.0 x86_64 i386

### What steps can reproduce the bug?

I have this dockerfile:

```
FROM oven/bun

WORKDIR /app

COPY package.json .
COPY bun.lockb .
COPY packages /app/packages

RUN bun install --production

COPY packages/api/src src
COPY packages/api/tsconfig.json .
# COPY public public

ENV NODE_ENV production
CMD ["bun", "src/index.ts"]

EXPOSE 3000
```

I use:

/ root
  /packages/api (bun server)
  /packages/ui
  
When I try to build the dockerfile, I get:

```
docker build -t api -f dockerfile .
[+] Building 1.6s (10/12)                                                                                                           docker:desktop-linux
 => [internal] load build definition from dockerfile                                                                                                0.0s
 => => transferring dockerfile: 567B                                                                                                                0.0s
 => [internal] load .dockerignore                                                                                                                   0.0s
 => => transferring context: 2B                                                                                                                     0.0s
 => [internal] load metadata for docker.io/oven/bun:latest                                                                                          0.0s
 => [1/8] FROM docker.io/oven/bun                                                                                                                   0.0s
 => [internal] load build context                                                                                                                   0.0s
 => => transferring context: 10.99kB                                                                                                                0.0s
 => CACHED [2/8] WORKDIR /app                                                                                                                       0.0s
 => CACHED [3/8] COPY package.json .                                                                                                                0.0s
 => CACHED [4/8] COPY bun.lockb .                                                                                                                   0.0s
 => CACHED [5/8] COPY packages /app/packages                                                                                                        0.0s
 => ERROR [6/8] RUN bun install --production                                                                                                        1.5s
------
 > [6/8] RUN bun install --production:
0.149 bun install v1.0.2 (37edd5a6)
1.146 error: IntegrityCheckFailed extracting tarball for source-map-support
1.162 error: IntegrityCheckFailed extracting tarball for @jridgewell/set-array
1.198 error: IntegrityCheckFailed extracting tarball for bson
1.259 error: IntegrityCheckFailed extracting tarball for @types/lodash.clonedeep
1.270 error: IntegrityCheckFailed extracting tarball for @vue/reactivity
1.292 error: IntegrityCheckFailed extracting tarball for @vue/devtools-api
1.310 error: IntegrityCheckFailed extracting tarball for terser
1.328 error: IntegrityCheckFailed extracting tarball for memory-pager
1.347 Segmentation fault
------
dockerfile:26
--------------------
  24 |     COPY packages /app/packages
  25 |
  26 | >>> RUN bun install --production
  27 |
  28 |     COPY packages/api/src src
 ```
 
 Oddly enough, this only started happening a few commits ago. I was able to deploy/build with this exact image previously.

### What is the expected behavior?

IntegrityCheckFailed shouldn't be happening :(

**Comment by jumoog** [comment]
So it worked with `oven/bun:1.0.1` ?

**Comment by ilrein** [comment]
It was working fine for 1.0.2, then I was developing the app merrily, and for some inexplicable reason it the docker build stopped working. I've been deleting node modules + lockfile but to no avail.

**Comment by ilrein** [comment]
Is there anyway to update the checksums? Seems like my project is dead right now :(

IE with yarn you could do `--update-checksums`

**Comment by ilrein** [comment]
I don't think this is a bun problem per se, it seems to be related to how you move node modules around in a project. If someone reads this in the future: make sure node_modules don't get put into the container.

**Comment by ghost** [comment]
> I don't think this is a bun problem per se, it seems to be related to how you move node modules around in a project. If someone reads this in the future: make sure node_modules don't get put into the container.

I have the exact same issue with Bun 1.0.16 (when installed without a `bun.lockb` file).
But the error only occurs when using `install` with the `--production` flag.

@ilrein What do you mean by "how you move node modules around in a project"? I basically do

```Dockerfile
FROM oven/bun:1.0

WORKDIR /app
COPY package.json package-lock.json ./
RUN bun install --frozen-lockfile --production
COPY src/ src/

ENTRYPOINT [ "bun" ]
```

So installing from `package-lock.json` should not cause the problem because
1. you used `bun.lockb` in your original issue and
2. it works without `--production`.

**Comment by cayter** [comment]
> > I don't think this is a bun problem per se, it seems to be related to how you move node modules around in a project. If someone reads this in the future: make sure node_modules don't get put into the container.
> 
> I have the exact same issue with Bun 1.0.16 (when installed without a `bun.lockb` file). But the error only occurs when using `install` with the `--production` flag.
> 
> @ilrein What do you mean by "how you move node modules around in a project"? I basically do
> 
> ```dockerfile
> FROM oven/bun:1.0
> 
> WORKDIR /app
> COPY package.json package-lock.json ./
> RUN bun install --frozen-lockfile --production
> COPY src/ src/
> 
> ENTRYPOINT [ "bun" ]
> ```
> 
> So installing from `package-lock.json` should not cause the problem because
> 
> 1. you used `bun.lockb` in your original issue and
> 2. it works without `--production`.

Bumped into the exact same issue... Sigh...

**Comment by cayter** [comment]
> > I don't think this is a bun problem per se, it seems to be related to how you move node modules around in a project. If someone reads this in the future: make sure node_modules don't get put into the container.
> 
> I have the exact same issue with Bun 1.0.16 (when installed without a `bun.lockb` file). But the error only occurs when using `install` with the `--production` flag.
> 
> @ilrein What do you mean by "how you move node modules around in a project"? I basically do
> 
> ```dockerfile
> FROM oven/bun:1.0
> 
> WORKDIR /app
> COPY package.json package-lock.json ./
> RUN bun install --frozen-lockfile --production
> COPY src/ src/
> 
> ENTRYPOINT [ "bun" ]
> ```
> 
> So installing from `package-lock.json` should not cause the problem because
> 
> 1. you used `bun.lockb` in your original issue and
> 2. it works without `--production`.

This is resolved by 1.0.20 which `--production` now installing all the dev dependencies.....
